common-utils/src/main/java/eu/arrowhead/common/mqtt/ArrowheadMqttService.java

```diff
@@ -113,4 +164,14 @@ public void response(
 			throw new ExternalServerError("MQTT service response failed: " + ex.getMessage());
 		}
 	}
+
+	//=================================================================================================
+	// assistant methods
+
+	//-------------------------------------------------------------------------------------------------
+	private String calculateConnectionId(final String address, final int port, final boolean isSSl) {
+		logger.debug("calculateConnectionId started...");
+
+		return new String(Base64.getDecoder().decode(address + port + String.valueOf(isSSl)), StandardCharsets.UTF_8);
```

Comment:
This is very strange. So you have a string, which is not a Base64 encoded string and then decode it?

Reply:
this was supposed to be an encoding...